pkg/controller/operators/operatorcondition_controller.go

```diff
@@ -252,6 +253,27 @@ func (r *OperatorConditionReconciler) ensureDeploymentEnvVars(operatorCondition
 	return nil
 }
 
+func (r *OperatorConditionReconciler) syncOperatorConditionStatus(operatorCondition *operatorsv2.OperatorCondition) error {
+	r.log.V(4).Info("Sync operatorcondition status")
+	currentGen := operatorCondition.ObjectMeta.GetGeneration()
+	changed := false
+	for _, cond := range operatorCondition.Spec.Conditions {
+		if c := meta.FindStatusCondition(operatorCondition.Status.Conditions, cond.Type); c != nil {
+			if cond.Status == c.Status && c.ObservedGeneration == currentGen {
+				continue
+			}
+		}
+		cond.ObservedGeneration = currentGen
+		meta.SetStatusCondition(&operatorCondition.Status.Conditions, cond)
+		changed = true
+	}
+
+	if changed {
+		return r.Client.Status().Update(context.TODO(), operatorCondition)
```

Comment:
Do we need to also log that we're attempting to update the `operatorcondition`, maybe at a higher verbosity? I guess it's possible that any encountered error will be logged at the call site, but may help down the line for debuggability.

Reply:
There is a log line above this to indicate that we are "syncing operatorcondition status" at level 4. That's significant. Are you recommending to log more info on this area?

Reply:
I was more thinking aloud about whether we'd want to log at a lower verbosity that we encountered a status condition (or conditions) that requires firing off an update request.